### ISSUE TITLE
$psEditor.Window.ShowErrorMessage never returns

### ISSUE THREAD

**Issue by SeeminglyScience** [PowerShell/PowerShellEditorServices#1106]
## Steps

1. In the integrated console run `$psEditor.Window.ShowErrorMessage('test')`
1. **Do not** click the <kbd>X</kbd> on the notifcation, instead click anywhere in the body and press escape.

## Expected

The notification to be dismissed and the prompt to return.

## Actual

The notification is dismissed, but the prompt never returns.  `ShowErrorMessage` blocks the pipeline indefinitely.

## Notes

The pipeline completes if the <kbd>X</kbd> is clicked instead.  All that said, I don't think it should ever wait for completion.  It feels real bad, especially from editor commands, to require dismissing the notification to free up the pipeline thread.

**Comment by TylerLeonhardt** [comment]
> . All that said, I don't think it should ever wait for completion.

I think that's been the behavior for a while now and I totally agree with you. It should be a notification, not a request expecting a response.

**Comment by SeeminglyScience** [comment]
> > . All that said, I don't think it should ever wait for completion.
> 
> I think that's been the behavior for a while now

Yeah I was bypassing it by sending the message directly with the old `IMessageSender`, since it had a general `waitForResponse` parameter.  New one doesn't have a way to do that afaict though.

**Comment by TylerLeonhardt** [comment]
https://github.com/PowerShell/PowerShellEditorServices/blob/master/src/PowerShellEditorServices/Services/PowerShellContext/EditorOperationsService.cs#L159-L180

A simple change. Instead of `SendRequest` it's `SendNotification`

Then in vscode-powershell change these to `NotificationType`:
https://github.com/PowerShell/vscode-powershell/blob/master/src/features/ExtensionCommands.ts#L142-L156

And change `onRequest` to `onNotification`
https://github.com/PowerShell/vscode-powershell/blob/master/src/features/ExtensionCommands.ts#L253-L267

**Comment by SeeminglyScience** [comment]
I think I'm just gonna use [`window/showMessage`](https://microsoft.github.io/language-server-protocol/specifications/specification-3-14/#window_showMessage) directly actually.  Still a good change to make though.

**Comment by andyleejordan** [comment]
Verified against current `main`: the vscode-powershell client handler now returns immediately — `showErrorMessage` does `void this.logger.writeAndShowError(message); return EditorOperationResponse.Completed;` (`src/features/ExtensionCommands.ts`) — so it no longer waits for the dialog to be dismissed, and `ShowErrorMessageAsync` returns promptly. The reported hang (prompt never returning after pressing Escape) no longer reproduces. Closing as completed. The separate nicety of making these `window/showMessage` notifications rather than requests (as discussed above) can be tracked separately if still wanted.

---
_Drafted by GitHub Copilot (Claude Opus 4.8) at @andyleejordan's request._
